src/ray/gcs/gcs_server/gcs_placement_group_manager.h

```diff
@@ -369,6 +370,25 @@ class GcsPlacementGroupManager : public rpc::PlacementGroupInfoHandler {
     usage_stats_client_ = usage_stats_client;
   }
 
+  /// Get a read only view of the pending placement groups.
+  ///
+  /// \return Pending placement groups.
+  const absl::btree_multimap<
+      int64_t,
+      std::pair<ExponentialBackOff, std::shared_ptr<GcsPlacementGroup>>>
+      &GetPendingPlacementGroups() const;
```

Comment:
Interesting... so we don't expose these infos before? How is the autoscaler currently working? Is this done in a push way inside placement group?

Reply:
Yeah right now it's weird. `GcsResourceManager` stores a `placement_group_load_` variable that `GcsPlacementGroupManager` pushes a value to periodically. 

[I saw there was a TODO not to stop doing that](https://github.com/ray-project/ray/blob/dc68ec95eaa3b16a350b0dce55485edd0c2bcc38/src/ray/gcs/gcs_server/gcs_resource_manager.h#L47), so I added this function instead.

Reply:
This is much cleaner.